### ISSUE TITLE
Port twisted.python.util.nameToLabel to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6158]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6158|
|Type|enhancement|
|Created|2012-10-23 15:56:30Z|
|Branch|https://github.com/twisted/twisted/tree/nameToLabel-py3-6158|

`nameToLabel` is required by `_sslverify.py`, so should be ported to Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6158 6158
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_nameToLabel_py3_6158 branches/nameToLabel-py3-6158
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1351007790000000 1351007790000000
changetime__1351081486000000 1351081486000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36248]) Branching to 'nameToLabel-py3-6158'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review, tests are running - http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/nameToLabel-py3-6158

This is a trivial port.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

`nameToLabel` seems to be missing some test coverage.  Add it now if you feel like it, otherwise file a ticket for addressing that.

Should we start a section in the porting howto that's just a list of APIs which took `str` before and changed to take `str` in Python 3?

Otherwise looks good, please merge when you're happy with respect to these two points.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

  1. [#6160](https://github.com/twisted/twisted/issues/6160)
  1. We already have some things, I thought, like `log.msg()`? In this particular case, however, the API documentation is very clear that it takes a "string that could be a Python identifier", so the behaviour on Python 3 should be clear to anyone who's used this API before.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set status to `closed`|
|-|-|

(In [36251]) Merge nameToLabel-py3-6158.

Author: itamar
Review: exarkun
Fixes: [#6158](https://github.com/twisted/twisted/issues/6158)

Port twisted.python.util.nameToLabel to Python 3.
